src/workers/cdc/primary.ts

```diff
@@ -736,10 +736,22 @@ const onSourcePostModerationChange = async (
   logger: FastifyBaseLogger,
   data: ChangeMessage<SourcePostModeration>,
 ) => {
-  if (data.payload.op === 'c' && !data.payload.after!.flags?.vordr) {
-    await triggerTypedEvent(logger, 'api.v1.source-post-moderation-submitted', {
-      post: data.payload.after!,
-    });
+  if (data.payload.op === 'c' && data.payload.after) {
+    try {
+      const flags = JSON.parse(data.payload.after.flags || '{}');
+
+      if (!flags?.vordr)
+        await triggerTypedEvent(
+          logger,
+          'api.v1.source-post-moderation-submitted',
+          { post: data.payload.after },
+        );
+    } catch (err) {
+      logger.error(
+        { post: data.payload.after },
+        'Error parsing flags for source post submission',
+      );
+    }
```

Comment:
I don't think we need catch here `JSON.parse(data.payload.after.flags || '{}');` will always resolve and if not something is really wrong so better that cdc fails and we catch it due to message not being processed

Reply:
Right. Yeah, we can remove the try-catch and logging.